src/fr-device/collector.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * @class Collector - base class for FRDevice
+ * Generic collector functions for collecting a device profile attributes
+ */
+class Collector {
+  /**
+   * @method reduceToObject - goes one to two levels into source to collect attribute
+   * @param props - array of strings; can use dot notation for two level lookup
+   * @param src - source of attributes to check
+   */
+  reduceToObject(props: string[], src: any): { [key: string]: string } {
```

Comment:
Is there no way to define `src`? It seems to be used as an array or an object in bracket notation.

Reply:
`src` is always an object, but its property values can vary wildly. The best I probably can do is this:

```
{
  [key: string]: any;
}
```

But, that's not much better.